src/swap.ts

```diff
@@ -0,0 +1,291 @@
+import { purry } from './purry';
+import { isString } from './isString';
+import { isArray } from './isArray';
+import { IterableContainer } from './_types';
+
+type isEqual<A, B> = A extends B ? (B extends A ? true : false) : false;
+
+type isLessThan<A extends number, B extends number> = isEqual<A, B> extends true
+  ? false
+  : 0 extends A
+  ? true
+  : 0 extends B
+  ? false
+  : isLessThan<Subtract<A, 1>, Subtract<B, 1>>;
+
+type BuildTuple<
+  L extends number,
+  T extends ReadonlyArray<unknown> = []
+> = T['length'] extends L ? T : BuildTuple<L, [...T, unknown]>;
```

Comment:
types usually describe the result, and not how they do it (they aren't functions). I think a better name for this would be TupleOfLength